core/store/src/trie/mem/arena/alloc.rs

```diff
@@ -1,17 +1,18 @@
 use near_o11y::metrics::IntGauge;
 
 use super::metrics::MEM_TRIE_ARENA_ACTIVE_ALLOCS_COUNT;
-use super::{ArenaMemory, ArenaSliceMut};
+use super::{ArenaMemory, ArenaPos, ArenaSliceMut};
 use crate::trie::mem::arena::metrics::{
     MEM_TRIE_ARENA_ACTIVE_ALLOCS_BYTES, MEM_TRIE_ARENA_MEMORY_USAGE_BYTES,
 };
+use crate::trie::mem::flexible_data::encoding::BorshFixedSize;
 
 /// Simple bump allocator with freelists. Allocations are rounded up to its
 /// allocation class, so that deallocated memory can be reused by a similarly
 /// sized allocation.
 pub struct Allocator {
-    freelists: [usize; NUM_ALLOCATION_CLASSES],
-    next_ptr: usize,
+    freelists: [ArenaPos; NUM_ALLOCATION_CLASSES],
+    next_ptr: ArenaPos,
```

Comment:
This field could benefit from a comment describing what this is for. `freelists` is kinda exlpained already by the ADT-level comment, but this is kinda unclear.

Reply:
Ah yeah these two both deserve comments. They were just missing altogether.